doc/developer/design/20220411_reclocking_implementation.md

```diff
@@ -0,0 +1,299 @@
+# Reclocking implementation
+
+## Summary
+
+This document proposes a design for implementing reclocking and exposing the
+reclocking mappings to end users.
+
+## Goals
+
+* Describe the physical representation of the reclocking data.
+* Describe the policy for reclocking each type of source.
+* Describe how users interact with the reclocking data.
+
```

Comment:
Is it worth adding an explicit note how this relates to `definite`ness of a source?